portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -273,6 +299,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect(REDIRECT_URL);
   }
 
+  /** 
+   * Returns a URL that points to the uploaded file, or null if the user didn't upload a file.
+   */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("comment-image");
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Check if the file is a valid image.
+    String filename = blobInfo.getFilename();
+    String[] splitFilename = filename.split("\\.");
+    String fileExtension = splitFilename[splitFilename.length - 1].toLowerCase();
```

Comment:
Can splitFileName be empty?

Reply:
I don't think it can. As long as filename is a String (even an empty String), filename will have the method split(); split() will return an array of String(s). If the splitting pattern is not found in filename, the returned array will just be an array of length 1, containing the full original String.